django/db/models/base.py

```diff
@@ -1608,6 +1608,22 @@ def _check_indexes(cls):
                         id='models.E034',
                     ),
                 )
+            if not (
+                connection.features.supports_partial_indexes or
```

Comment:
You can't use `connection` here, you need to pass `databases` to `_check_indexes` and follow what `_check_constraints` does.

Reply:
Hmm, seems like I need to separate the check into another function.